prac_09/taxi.py

```diff
@@ -0,0 +1,32 @@
+"""
+CP1404/CP5632 Practical
+Car class
+"""
+from prac_09.car import Car
+
+
+class Taxi(Car):
+    """Specialised version of a Car that includes fare costs."""
+    price_per_km = 1.23
+    def __init__(self, name, fuel):
+        """Initialise a Taxi instance, based on parent class Car."""
+        super().__init__(name, fuel)
+        self.current_fare_distance = 0
+
+    def __str__(self):
+        """Return a string like a Car but with current fare distance."""
+        return f"{super().__str__()}, {self.current_fare_distance}km on current fare, ${self.price_per_km:.2f}/km"
+
+    def get_fare(self):
+        """Return the price for the taxi trip."""
+        return self.price_per_km * self.current_fare_distance
+
+    def start_fare(self):
+        """Begin a new fare."""
+        self.current_fare_distance = 0
+
+    def drive(self, distance):
```

Comment:
I approached this a different way and got the matching sample output. I tried it with yours and got a different output for the prices. Maybe that's not due to that exact method but how you have programmed other aspects but I'd recommend double checking this output. :)